Windows dark theme added

0x4A4D00 opened the pull request:
Hello,
I added the [metadarkstyle ](https://github.com/zamtmn/metadarkstyle) package, so now we have a dark theme on Windows too.

![LazScope 3_9_2025 5_30_41 PM](https://github.com/user-attachments/assets/b449660c-17eb-48d7-bd69-fd4e1a66d69a)

0x4A4D00 committed:
Windows dark theme added

ccrause commented:
Thank you for this PR, it looks very nice.

To include this into the source some changes are required:
* The metadarkstyle package does not ship with a default install of Lazarus. Check if the metadarkstyle package can be added as a submodule so that the required files are automatically included when this repository is checked out.
* This only works on Windows, so the package needs to be excluded from the project when building on other platforms such as Linux/GTK etc.

0x4A4D00 committed:
Windows dependent build fixed - metadarkstyle submodule added

0x4A4D00 commented:
Hello, thank you for your feedback!
I have addressed both points: I added the metadarkstyle package as a submodule, and I added a compiler macro to ensure that it is excluded from the project when building on non-Windows platforms.

ccrause commented:
Thanks, the build works on my Linux machine again.

Last question (hopefully :smiley:): can the dark theme be changed at runtime to other themes?  I haven't used this package yet and I know not everyone prefers a dark theme, so can one provide a runtime option to switch to different styles?  If not, I will have to create build modes to generate release binaries with and without the dark theme, so not a big problem.

0x4A4D00 commented:
Yes, I set the 'PreferredAppMode' to 'pamAllowDark,' which allows the theme to change according to the Windows theme. This way, users can switch between dark and light themes based on their OS theme.

ccrause commented:
OK, thank you.
